src/ast/query.rs

```diff
@@ -528,6 +533,46 @@ impl fmt::Display for ExceptSelectItem {
     }
 }
 
+/// Bigquery `REPLACE` information.
+///
+/// # Syntax
+/// ```plaintext
+/// REPLACE <new_expr> AS <col_name>
+/// ```
+#[derive(Debug, Clone, PartialEq, PartialOrd, Eq, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
+pub enum ReplaceSelectItem {
```

Comment:
Two questions about this structure.

1. Why don't you have an enum with two fields, an expression (the <expression> field) and another with an identifier? 
2. Any reason why you put the `AS` as mandatory? BigQuery has it as non-mandatory, but this syntax you displayed here says otherwise.

If you have a select item, the user will have to validate that the select item is not an expression or wildcard, which isn't excellent as we are such that it isn't.
- You'll have to validate the column name expression, as it MUST be an identifier.

Reply:
1. That make sense. It's true that having a select item needs additional validation. I'll modify the implementation.
2. I think `AS` is required in `SELECT * Replace` statement. This is the special case.
https://cloud.google.com/bigquery/docs/reference/standard-sql/query-syntax#select_replace

Reply:
@togami2864 check the syntax structure for the select statement: https://cloud.google.com/bigquery/docs/reference/standard-sql/query-syntax?hl=pt-br#select_list

```
select_all:
    [ expression. ]*
    [ EXCEPT ( column_name [, ...] ) ]
    [ REPLACE ( expression [ AS ] column_name [, ...] ) ]
```

`AS` is optional between the expression and its column, if I understood it correctly.

Reply:
Sorry, my bad. Required element is `expr` and `column_name`.